scheduler/scheduler-server/src/test/java/functionaltests/job/log/TestPreciousLogs.java

```diff
@@ -188,6 +190,32 @@ private void testPreciousLogs(boolean createJavaTask, boolean forkEnv, boolean g
                                     expectedLine, expectedOccurrences, StringUtils.countMatches(output, expectedLine));
             }
         }
+
+        // Test log removal
+        schedulerHelper.removeJob(jobId);
+
+        waitUntilAllLogsWereRemoved(jobId, logFiles);
+    }
+
+    private void waitUntilAllLogsWereRemoved(JobId jobId, List<File> logFiles) throws InterruptedException {
+        int cpt = 0;
+        while (!allLogFilesRemoved(logFiles) && cpt < 200) {
+            Thread.sleep(100);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)